lib/vagrant-libvirt/provider.rb

```diff
@@ -69,12 +69,11 @@ def ssh_info
         }
 
         if @machine.provider_config.connect_via_ssh
-          ssh_info[:proxy_command] =
-            "ssh '#{@machine.provider_config.host}' " \
-            "-l '#{@machine.provider_config.username}' " \
-            "-i '#{@machine.provider_config.id_ssh_key_file}' " \
-            'nc %h %p'
-
+          proxy_command = "ssh '#{@machine.provider_config.host}' "
+          proxy_command << "-l '#{@machine.provider_config.username}' " if @machine.provider_config.username
+          proxy_command << "-i '#{@machine.provider_config.id_ssh_key_file}' " if @machine.provider_config.id_ssh_key_file
+          proxy_command << '-W %h:%p'
+          ssh_info[:proxy_command] = proxy_command
```

Comment:
General recommendation is to try and keep individual fixes separate. I've been trying to think of how to test this, and I don't think there is an easy way to add unit tests other than simply inspecting ssh_info returned based on certain inputs, but I think it makes sense so worth  just merging as is. Probably best to provide a separate PR for this and that can be merged just in case we need to look at the above config changes again after being updated.